docs/source/column.rst

```diff
@@ -41,11 +60,58 @@ NumericalColumn class reference
     NumericalColumn.round
     NumericalColumn.log
 
+    NumericalColumn.describe
     NumericalColumn.min
     NumericalColumn.max
     NumericalColumn.sum
     NumericalColumn.mean
     NumericalColumn.std
+    NumericalColumn.median
+
+
+StringColumn class reference
+-----------------------------------
+.. class:: StringColumn()
 
+.. autosummary::
+    :toctree: generated
+    :nosignatures:
+
+    torcharrow.istring_column.StringMethods.length
```

Comment:
tried to use `StringColumn.str.length` but seems Sphinx doesn't work. @NivekT should I just use `istring_column.StringMethods.length` and some how do string substitution in doc?

Reply:
Maybe try this?
https://stackoverflow.com/questions/20569011/python-sphinx-autosummary-automated-listing-of-member-functions

Reply:
It should allow you to show methods under class. Let me know if that helps

Reply:
@NivekT I mean, the problem is `StringColumn.str` is a member in `StringColumn`~

Reply:
How about using alias in TA?
`torcharrow.StringColumn.str.length = torcharrow.istring_column.StringMethods.length`

Reply:
Yea in that case, the substitution that you linked may be the most straightforward way.

Reply:
here the substitution only changes the signature right? We can't make the change to function name

Reply:
How does it currently look and how would you like it to be?

You might get a shorter name if you change `.. currentmodule::` to something else.

Reply:
Currently it looks like this:
<img width="853" alt="Screen Shot 2022-06-16 at 10 37 34 AM" src="https://user-images.githubusercontent.com/745238/174132605-caf8045c-c2f5-44c1-bd9e-44c86eeb536a.png">

We want to shorten the `torcharrow.istring_column.StringMethods` part to `StringColumn.str.length`, is that right? @wenleix . Since this is the function name not signature, the process-signature hook won't help here.

Reply:
Can you create an alias in python using import statement then show the aliases in the doc?

For example, in `TorchData` we do the following:
```python
from torch.utils.data.datapipes.iter.callable import (
    CollatorIterDataPipe as Collator,
    MapperIterDataPipe as Mapper,
)
```
It shows up as this in our doc:

![Screen Shot 2022-06-16 at 3 09 24 PM](https://user-images.githubusercontent.com/4935152/174146874-e1151b83-81e9-4a6c-ab66-378d94d06ed2.png)

Reply:
> ```python
> from torch.utils.data.datapipes.iter.callable import
> ```

where is this code located?

Reply:
https://github.com/pytorch/pytorch/blob/master/torch/utils/data/datapipes/iter/__init__.py

Reply:
Also this part within `TorchData`'s`conf.py` may be worth looking into:

```python
def setup(app):

    # Overwrite class name to allow aliasing in documentation generation
    import torchdata.datapipes.iter as iter
    import torchdata.datapipes.map as map

    for mod in (iter, map):
        for name, obj in mod.__dict__.items():
            if isinstance(obj, type):
                obj.__name__ = name

    app.connect("autodoc-process-signature", process_signature)
```